packages/frontend/src/components/composer/Composer.tsx

```diff
@@ -137,6 +138,17 @@ const Composer = forwardRef<
     }
   }, [selectedChat.canSend])
 
+  useEffect(() => {
+    if (
+      currentComposerMessageInputRef.current?.getText() === '' &&
+      !draftState.file
+    ) {
+      setShowSendButton(false)
+    } else {
+      setShowSendButton(true)
+    }
+  }, [currentComposerMessageInputRef, draftState])
```

Comment:
This looks fragile to me. The effect only runs when `draftState` changes, but it changes only when we save the draft to the back-end.
Additionally, `showSendButton` also affects the button when it's an "edit message" button.

Does it have to be an effect? `showSendButton` is basically a computed value, and not a state of its own.
Can't we just

```typescript
const showSendButton =
  currentComposerMessageInputRef.current?.getText() !== '' ||
  draftState.file != null`?
```

Reply:
This would also solve the

>  Only downside so far: it does not work when a message is edited, but we have a warning there if the send button is clicked without a message text

Reply:
Hmm, I just realized that `currentComposerMessageInputRef.current?.getText()` will still only run on the re-renders of this component, because the state of the text input is not exposed to it. Maybe it's time to lift the state of the text input?

Either way what I suggested above should be an improvement, unless it somehow doesn't work.

Reply:
>  Can't we just
> 
> ```ts
> const showSendButton =
>   currentComposerMessageInputRef.current?.getText() !== '' ||
>   draftState.file != null
> ```

As you mentioned already that does not hide the button after sending the message since there is no rerender in the Composer component.

> The effect only runs when draftState changes, but it changes only when we save the draft to the back-end.

Yes, but that happens after each change so it is a reliable trigger I would say

But anyway I will investigate for more possibilities. As said above this is a minimistic approach so far

Reply:
Even if rewriting this without an effect doesn't improve behavior, I think it's better because it's simpler code-wise.

Reply:
I kept the useEffect since we need the changes from ComposerInput to trigger a render as said above. I included now also the edit text scenario.

This is as said an approach to avoid much refactoring since we will do that "later" with the Composer refactoring.

But of course if you have a better idea you are welcome to propose a MR :-)

Reply:
With the latest commit the "send" button doesn't disappear after you send a message.

If we really want this behavior, I suggest to revert that last commit and merge. I tested the MR prior to that commit and it seemed to work alright.

Reply:
> With the latest commit the "send" button doesn't disappear after you send a message.

So removing the text from the textfield does not trigger the onChange event. It would need another onChange call  
immediately after sending the message:
```
this.props.sendMessageOrEditRequest()
this.props.onChange('')
```

Reply:
It has to do with the fact that `textareaRef.setText()` is used outside of the component, I think it's async after the message gets sent, in `useDraft`.

Reply:
Perhaps as another temporary hack one could add `onChange` to inside of the `setText()` method of the `ComposerTextarea` component.